template/web/app/paymaster-bundler/_utils/fetchNFTs.ts

```diff
@@ -0,0 +1,60 @@
+import { SmartAccountClient } from 'permissionless';
+import { PublicClient } from 'viem';
+
+import { nftAbi } from '../_components/abi';
+import { NFTType } from '../types';
+
+export default async function fetchNFTs(smartAccount: SmartAccountClient, client: PublicClient) {
+  // Get # of NFTs owned by address
+  const address = smartAccount.account?.address;
+
+  const numTokens = await client.readContract({
+    address: '0x66519FCAee1Ed65bc9e0aCc25cCD900668D3eD49',
+    abi: nftAbi,
+    functionName: 'balanceOf',
+    args: [address],
+  });
+
+  // Get the token IDs and metadata by token ID
+  var tokens = [] as string[];
+
+  for (let i = 0; i < Number(numTokens); i++) {
+    const tokenID = await client.readContract({
+      address: '0x66519FCAee1Ed65bc9e0aCc25cCD900668D3eD49',
+      abi: nftAbi,
+      functionName: 'tokenOfOwnerByIndex',
```

Comment:
Worth noting this approach will only work with contracts that support IERC721Enumerable.. might confuse users why this won't work with a standard ERC721 contract.